docs/specs/account-management-api.md

```diff
@@ -0,0 +1,861 @@
+}
+```
+
+Response
+
+If verification is not required, then the email is added immediately.
+
+```json
+{
+  "result": {
+    "identification_method": {
+      "identification": "email",
+      "login_id": "user@example.com",
+      "claims": {
+        "email": "user@example.com"
+      },
+      "created_at": "2006-01-02T03:04:05Z",
+      "updated_at": "2006-01-02T03:04:05Z"
+    }
+  }
+}
+```
+
+If verification is required, you need to perform verification.
+
+```json
+{
+  "result": {
+    "verification": {
+      "token": "verificationtoken_blahblahblah",
+      "channel": "email",
+      "otp_form": "code",
+      "code_length": 6,
+      "can_resend_at": "2023-09-21T00:00:00+08:00",
+      "can_check": false,
+      "failed_attempt_rate_limit_exceeded": false
+    }
+  }
+}
+```
+
+Use [Verify OTP](#verify-otp) and [Resend OTP](#resend-otp) to continue.
+
+If the email is already taken by another account, then you will receive the following error.
+
+```json
+{
+  "error": {
+    "name": "Invalid",
+    "reason": "InvariantViolated",
+    "message": "identity already exists",
+    "code": 400,
+    "info": {
+      "cause": {
+        "kind": "DuplicatedIdentity"
+      }
+    }
+  }
+}
+```
+
+### Add a new phone number, with verification
+
+`POST /api/v1/account/identification`
+
+Request
+
+```json
+{
+  "identification": "phone",
+  "login_id": "+85251000001"
+}
+```
+
+Response
+
+If verification is not required, then the phone number is added immediately.
+
+```json
+{
+  "result": {
+    "identification_method": {
+      "identification": "phone",
+      "login_id": "+85251000001",
+      "claims": {
+        "phone_number": "+85251000001"
+      },
+      "created_at": "2006-01-02T03:04:05Z",
+      "updated_at": "2006-01-02T03:04:05Z"
+    }
+  }
+}
+```
+
+If verification is required, you need to perform verification.
+
+```json
+{
+  "result": {
+    "verification": {
+      "token": "verificationtoken_blahblahblah",
+      "channel": "sms",
+      "otp_form": "code",
+      "code_length": 6,
+      "can_resend_at": "2023-09-21T00:00:00+08:00",
+      "can_check": false,
+      "failed_attempt_rate_limit_exceeded": false
+    }
+  }
+}
+```
+
+Use [Verify OTP](#verify-otp) and [Resend OTP](#resend-otp) to continue.
+
+If the phone number is already taken by another account, then you will receive the following error.
+
+```json
+{
+  "error": {
+    "name": "Invalid",
+    "reason": "InvariantViolated",
+    "message": "identity already exists",
+    "code": 400,
+    "info": {
+      "cause": {
+        "kind": "DuplicatedIdentity"
+      }
+    }
+  }
+}
+```
+
+### Add a new username
+
+`POST /api/v1/account/identification`
+
+Request
+
+```json
+{
+  "identification": "username",
+  "login_id": "johndoe"
+}
+```
+
+Response
+
+```json
+{
+  "result": {
+    "identification_method": {
+      "identification": "username",
+      "login_id": "johndoe",
+      "claims": {
+        "preferred_username": "johndoe"
```

Comment:
actually in what cases would the `claims` be different from the `login_id`?

Reply:
No case. Just add `claims` so that the developer can just read from the `claims` to get the `email`, `phone_number`, etc. Otherwise, they have to switch on `identification` and determine the login ID is actually a email, phone number or a username.

Reply:
ok